test/upgrade/prober/wathola/sender/services.go

```diff
@@ -50,17 +55,31 @@ func (s *sender) SendContinually() {
 		close(shutdownCh)
 	}()
 
+	var start time.Time
+	retry := 0
 	for {
 		select {
 		case <-shutdownCh:
+			// If there is ongoing event transition, we need to push to retries too.
+			if retry != 0 {
+				s.unavailablePeriods = append(s.unavailablePeriods, time.Since(start))
+			}
 			return
 		default:
 		}
 		err := s.sendStep()
 		if err != nil {
-			log.Warnf("Could not send step event, retry in %v", senderConfig.Cooldown)
-			time.Sleep(senderConfig.Cooldown)
+			if retry == 0 {
+				start = time.Now()
+				log.Warnf("Could not send step event %v, retrying", s.eventsSent)
+			}
+			retry++
 		} else {
+			if retry != 0 {
+				s.unavailablePeriods = append(s.unavailablePeriods, time.Since(start))
+				log.Warnf("Event sent after %v retries", retry)
+				retry = 0
+			}
 			time.Sleep(senderConfig.Interval)
```

Comment:
After removing Cooldown, this sleep needs to be moved to outside of `if` statement, as now we are sleeping only after successful send

Reply:
I actually did that on purpose. So that when it is retrying it is always sending as fast as possible to get the most accurate down time. The request and response takes time long enough to make the CPU cool anyway.

Reply:
Okay. Makes sense. If that will be making problems we can always move that sleep.